### DISCUSSION TITLE
Changing master port from 5000 to 5432

### DISCUSSION THREAD

**Discussion by klention** [autobase-tech/autobase#119] [General]
Hello,
First, thank you @vitabaks and anyone else supporting on this project.
My question is an easy one, Can I change the port where master is listening from 5000 to 5432?
I understand that in order to do this one PostgreSQL has listen on another one beside 5432 and pgbouncer has to send request to the new port.
I didn't manage to find something regarding this one on the documentations. If you can guide me how, it will be great.
Have a great day and thank you!
Klention

**Comment by vitabaks** [comment]
Hello @klention 

**For Type B scheme (without load balancing):**
It is enough to specify the postgresql_port variable to any other port, for example 5433, and specify pgbouncer_listen_port: 5432

**For Type A scheme (with load balancing - haproxy):**
I added `haproxy_listen_port` variables, please download the playbook again.

1. Specify the postgresql_port variable to any other port, for example 5433
2. Specify haproxy_listen_port.master: 5432

For an already deployed cluster, you can execute playback with tag to update the haproxy config, haproxy template for confd, and pgbouncer config file:
`ansible-playbook deploy_pgcluster.yml --tags haproxy_conf,confd_conf,pgbouncer_conf`

**Comment by klention** [comment]
Hello @vitabaks,
Thanks for the feedback. I did what you wrote and the port where PostgreSQL and haproxy were listening (read-write node) changed accordingly, but unfortunately I could not connect to it. I can connect to the other ones (replica nodes).
What I ended up doing was changing every port value "5000" what I found inside playbooks into 5432 and that worked.
Thank you and have a great day!
Klention

**Reply by vitabaks** [reply]
> but unfortunately I could not connect to it. I can connect to the other ones (replica nodes).

For sure, I missed. You also needed to correct the port in the variable `firewall_allowed_tcp_ports_for.balancers`

fixed: 3a0a036f8419a6279a21b992337eb5467d54db5f

**Comment by klention** [comment]
Thank you! It works like a charm.
Have a great day!
